src/provider/provider_os_memory.c

```diff
@@ -248,35 +248,66 @@ static umf_result_t translate_params(umf_os_memory_provider_params_t *in_params,
         return result;
     }
 
-    provider->fd = os_create_anonymous_fd(provider->visibility);
-    if (provider->fd == -1) {
-        LOG_ERR(
-            "creating an anonymous file descriptor for memory mapping failed");
-        return UMF_RESULT_ERROR_UNKNOWN;
-    }
+    // size_fd will be increased during each allocation if (provider->fd > 0)
+    provider->size_fd = 0;
+    provider->shm_name[0] = 0; // zero shm_name
+
+    if (in_params->visibility != UMF_MEM_MAP_SHARED) {
+        provider->fd = -1;
+        provider->max_size_fd = 0;
+    } else if (in_params->shm_name) {
+        provider->fd = os_shm_open(in_params->shm_name);
```

Comment:
I would refactor this function due to the following concerns:

- The function is too long now. Need to put some code blocks into separate functions.
- The name of the function `translate_params` is misleading. This function not only translates parameters but also open shared memory file or creates anonymous FD, etc.

Reply:
Done